### PR TITLE
Mon 3581 compatibility strict mode

### PR DESCRIPTION
**Pull request by loiclau**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreon-clapi/centreonLDAP.class.php [hunk 1]
```diff
@@ -215,14 +216,13 @@ public function add($parameters)
...
-        $this->db->query(
-            "INSERT INTO auth_ressource (ar_name, ar_description, ar_enable) VALUES (:name, :description, :status)",
-            array(
-                ':name' => $name,
-                ':description' => $description,
-                ':status' => '1'
-            )
+        $stmt = $this->db->prepare(
+            "INSERT INTO auth_ressource (ar_name, ar_description, ar_enable, ar_type)
+                VALUES (:arName, :description, '1', '')"
```
Comment (sc979) [inline www/class/centreon-clapi/centreonLDAP.class.php]:
```suggestion
            VALUES (:arName, :description, '1', '')"
```

File: www/install/createTables.sql [hunk 2]
```diff
@@ -242,7 +242,7 @@ CREATE TABLE `auth_ressource` (
   `ar_id` int(11) NOT NULL AUTO_INCREMENT,
   `ar_name` varchar(255) NOT NULL DEFAULT 'Default',
   `ar_description` varchar(255) NOT NULL DEFAULT 'Default description',
-  `ar_type` varchar(50) NOT NULL,
+  `ar_type` varchar(50) NOT NULL DEFAULT 'Default type',
```
Comment (sc979) [inline www/install/createTables.sql]:
```suggestion
  `ar_type` varchar(50) NOT NULL DEFAULT 'ldap',
```

File: www/install/sql/centreon/Update-DB-19.10.0.sql [hunk 3]
```diff
@@ -53,6 +53,7 @@ ALTER TABLE `traps_service_relation` DROP COLUMN `tsr_id`;
 --
 ALTER TABLE `acl_groups` MODIFY COLUMN `acl_group_changed` int(11) NOT NULL DEFAULT 1;
 ALTER TABLE `widget_models` MODIFY COLUMN `description` TEXT NOT NULL;
+ALTER TABLE `auth_ressource` ALTER `ar_type` SET DEFAULT 'Default type';
```
Comment (sc979) [inline www/install/sql/centreon/Update-DB-19.10.0.sql]:
```suggestion
ALTER TABLE `auth_ressource` ALTER `ar_type` SET DEFAULT 'ldap';
```

**Comment by sc979** [comment]
Please fill the template too ;)

File: www/class/centreon-clapi/centreonLDAP.class.php [hunk 4]
```diff
@@ -49,6 +49,7 @@
 class CentreonLDAP extends CentreonObject
 {
     protected $db;
+    protected $baseParams;
```
Comment (jdelpierre) [inline www/class/centreon-clapi/centreonLDAP.class.php]: What you add this ?

Comment (loiclau) [reply www/class/centreon-clapi/centreonLDAP.class.php]: a no initialised variable use lign 69

**Pull request merged**